### ISSUE TITLE
Drow ranger's attack range is incorrect

### ISSUE THREAD

**Issue by Stephanie55** [ValveSoftware/Dota2-Gameplay#2924]
### Ability name

Frost arrows

### Description

![20220909011721_1](https://user-images.githubusercontent.com/40859719/189242170-c4457be3-8bf2-4433-95e5-4baed044db67.jpg)
![20220909011726_1](https://user-images.githubusercontent.com/40859719/189242173-ba176527-544d-45c2-b555-6e7f0a579fb8.jpg)
![20220909011732_1](https://user-images.githubusercontent.com/40859719/189242177-ed88e6c8-29b6-4ec8-9616-2100a98b5a9c.jpg)

Her real attack range is 650. It was never mentioned in a patch and the tool tip says 625. Her frost arrows' range however is 625.

### Example Match ID (and possibly Timestamp)

Demo mode

**Comment by TheUnholyWrath** [comment]
Its Collision Size. Drow's "real" attack range is 649. Its most obvious to see on Towers; they have 700 attack range and 144 collision size so they can hit units 844 away from their center. Idk if theres a bug in here somewhere, seems like its intentional but explained nowhere.

**Comment by dobacetr** [comment]
I think the attack range should be computed from the center of the hero rather than the closest collision point.
I believe it was last year when the Dragon Knight persona was released, it was discovered that the Fire Breath originates from DK's mouth rather than where the hero is. Persona having that "bone" at the different point resulted in different skill outcomes depending on the cosmetics, which is 100% unintended.
In short, game mechanics should not rely on models or collisions for these kind of computations. A collision change (or possibly a new skill/item) would also affect range of the hero in an indirect way. I don't think this is a good interaction of mechanics.

**Comment by Stephanie55** [comment]
The problem that Q cast range and drow's attack range are different. They should be both 625.

**Comment by dobacetr** [comment]
Thank you for the clarification.

This is Drow Ranger at her maximum attack range.
![image](https://user-images.githubusercontent.com/97127808/189445181-97f9a2e6-095b-4112-b9bf-ba7d779ba5bd.png)

When you manually use _Frost Arrows_ to target the enemy, Drow Ranger actually takes a few steps forward to fire her shots.
![image](https://user-images.githubusercontent.com/97127808/189445263-390d1ce2-3903-4c98-a41c-0c65db523865.png)

Note that, auto-cast is proc'ed on maximum attack range as well. It is only when the _Frost Arrows_ are casted manually that the Drow Ranger needs to move forward. So, this is indeed a bug.

Outworld Destroyer has the exact same problem with _Arcane Orb_.
Clinkz has the exact same problem with _Searing Arrows_.
Weaver has the exact same problem with _Geminate Attack_.
Ancient Apparition has the exact same problem with _Chilling Touch_.
Enchantress has the exact same problem with _Impetus_.
Silencer has the exact same problem with _Glaives of Wisdom_.

Huskar has this problem in reverse. _Burning Spear_ has further range than his auto-attacks.

Viper does not have this problem.

**Comment by Aluerie** [comment]
> The problem that Q cast range and drow's attack range are different. They should be both 625.

No, attack range starts from `bounding_range` of the hero which is 25 for 99% of heroes while cast range starts from the center of the hero. Currently they are both 625 for drow - and this is why the problems occurs. 

---

Also,  I don't really want to make a new issue for any of these but 

* Enchanted Quiver Neutral Item both passive bonus attack range and bonus range from Damage proc malfunction with manual cast of orb-effects.
* Drow new shard bonus attack range malfunction with manual orb-cast #8946
* Viper also has a problem. Normal attack range 575. Manual cast range of level 1 Q: 600 (correct). But autocast-on uses 600 as attack range leading to again extra 25 distance.

**Comment by dobacetr** [comment]
> > The problem that Q cast range and drow's attack range are different. They should be both 625.
> 
> No, attack range starts from the `bounding_range` radius edge of the hero which is 25 for 99% of heroes while cast range starts from the center of the hero. Currently they are both 625 for drow - and this is why the problems occurs.
> ...

That is exactly what I was alluding to. I think that attack range should start from the center.

**Comment by EpsilonD3lta** [comment]
AA (lvl 30 + shard + agh) with dragon lance has longer attack range when chilling touch is casted manually and shorter range when on autocast. Without dragon lance it has equal range.

**Comment by Aphotic** [comment]
[new reddit thread](https://old.reddit.com/r/DotA2/comments/170k470/drow_rangers_frost_arrows_cast_range_is_always/)
